plugins/connection/incus.py

```diff
@@ -104,11 +104,17 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
         if getattr(self._shell, "_IS_WINDOWS", False):
             # Initializing regular expression patterns to match on a PowerShell or cmd command line.
             self.powershell_regex_pattern = re.compile(
-                r"^(?P<executable>(\"?([a-z]:)?[a-z0-9 ()\\.]+)?powershell(\.exe)?\"?|(([a-z]:)?[a-z0-9()\\.]+)?powershell(\.exe)?)\s+.*(?P<command>-c(ommand)?)\s+",  # noqa: E501
+                (
+                    r"^(?P<executable>("
+                    r'"([a-z]:)?[a-z0-9 ()\\.]+)?powershell(\.exe)?"'  # "c:\powershell.exe" or "\\host\powershell.exe"
+                    r"|"
+                    r"([a-z]:)?[a-z0-9()\\.]+)?powershell(\.exe)?"  # same thing, no quotes, and no whitespace
+                    r")\s+.*(?P<command>-c(ommand)?)\s+"
+                ),
```

Comment:
And something else that eluded me before. This:
```
"c:\powershell.exe" -c
```
would match the "first half" (with quotes) and it would assign `"c:\powershell.exe"` to the match group `executable`. But this:
```
c:\powershell.exe -c
```
would match the "second half" (w/out quotes), and the it would be _outside the capturing group_, so `executable` would come empty!

So, I would suggest we simplify it further to:
```suggestion
                r'^"?(?P<executable>(?:[a-z]:)?[a-z0-9 ()\\.]*powershell(?:\.exe)?)"?\s+.*(?P<command>-c(?:ommand)?)\s+',
```
This way, the regexp:
- keeps the unbalanced quotes issue (but that should not be a major problem as the code below, in `_build_command()` actually `.strip('"')` the string anyways), 
- don't repeat your/itself
- supports the case `c:powershell.exe`, by changing the `+` into `*` right before `powershell`
- change the other, unnamed, groups into non-capturing groups, by using `(?: )` instead of `( )`

Reply:
Adding a mandatory backslash in front of powershell will make the use case `powershell -c my_command` not matching, which is valid in all generic cases.

If we really want to, we could add the literal in the drive letter `c:\`:
`^"?(?P<executable>(?:[a-z]:\\)?[a-z0-9 ()\\.]*powershell(?:\.exe)?)"?\s+.*(?P<command>-c(?:ommand)?)\s+`

This should make these valid:
```
c:\powershell.exe
\\127.0.0.1\share\powershell
\\127.0.0.1\share\powershell.exe
powershell
powershell.exe
c:\mypath\powershell
c:\mypath\powershell.exe
c:\my path\powershell
c:\my path\powershell.exe
"c:\powershell.exe"
"\\127.0.0.1\share\powershell"
"\\127.0.0.1\share\powershell.exe"
"powershell"
"powershell.exe"
"c:\mypath\powershell"
"c:\mypath\powershell.exe"
"c:\my path\powershell"
"c:\my path\powershell.exe"

And many more..
```

Reply:
That makes perfect sense. I have up added the extra `\\` to both powershell and cmd regexps.